### ISSUE TITLE
Sequence Classification

### ISSUE THREAD

**Issue by shoaibahmed** [Element-Research/rnn#94]
Thanks a lot for developing such an awesome package.
I am trying to achieve human activity recognition task using BDLSTM. Can you pl provide me with a code snipped that implements sequence classification task (assigning a single label to whole sequence) using BDLSTM?

**Comment by nicholas-leonard** [comment]
@shoaibahmed Should look somewhat like this:

```lua
blstm = nn.Sequential()
   :add(nn.BiSequencer(nn.LSTM(inputSize, outputSize)))
   :add(nn.SelectTable(-1))
   :add(nn.Linear(outputSize*2))
   :add(nn.LogSoftMax(nActivity))
```

Note : I haven't debugged it.

**Comment by shoaibahmed** [comment]
Can you pl give a snippet regarding how to back-propagate the gradient in this case?

**Comment by nicholas-leonard** [comment]
@shoaibahmed 

```lua
nll = nn.ClassNLLCriterion()

-- batchSize sequences of 2 time-steps each
input = {torch.randn(batchSize, inputSize), torch.randn(batchSize, inputSize)}
-- target is random class between 1 and nActivity
target = torch.LongTensor(batchSize):random(1,nActivity)

output = brnn:forward(input)
err = nll:forward(output, target)

gradOutput = nll:backward(output, target)
brnn:backward(input, gradOutput)
```

**Comment by shoaibahmed** [comment]
Thanks a lot @nicholas-leonard. That was really helpful.
I was unable to figure out how to back-propagate just a single label for the whole sequence. Your snippet made it clear.

Thanks once again for your time.

**Comment by shoaibahmed** [comment]
Is this RNN package supported on Android?

**Comment by nicholas-leonard** [comment]
@shoaibahmed It's not by default part of the torch-android package, but getting it to work on the phone shouldn't be too difficult. We did it internally here at Element Inc. Worked fine.
